### PR TITLE
Workaround for OpenJ9 issue #2742

### PR DESCRIPTION
**Pull request by aguibert**
#### Short description of what this resolves:

I've been testing out Liberty with OpenJ9 JDK 11 builds and found that VM attach isn't fully working.  There is already an issue open in OpenJ9 to track this here: https://github.com/eclipse/openj9/issues/2742

### CODE REVIEW

**Comment by Azquelt** [comment]
I'm not quite sure how this fix lines up with the OpenJ9 defect. That defect talks about being unable to use the Attach API to load and start the management agent as a normal agent, requiring the use of `startLocalManagementAgent()`.

However, the existing code doesn't try to start the management agent at all. We expect it to have already been started by the liberty server and the `localConnector-1.0` feature.

**Comment by Azquelt** [comment]
Also, please fix the whitespace.

**Comment by aguibert** [comment]
@Azquelt the issue seemed similar because Arquillian could connect to the Liberty VM, but it was not reading the jmx address property.  Calling `startLocalManagementAgent()` was a workaround to have that property get initialized.

I've amended my commit to use a different approach this time.  Instead of trying to read the property from the `VirtualMachine` object, we can attempt to read the JMX address file that Liberty outputs at `$WLP_OUTPUT_DIR/workarea/` when the `localConnector-1.0` feature is enabled.

**Comment by Azquelt** [comment]
Hmm, not sure about the workarea, but you got me investigating further.

The KC docs seem to suggest using `${server.output.dir}/logs/state/com.ibm.ws.jmx.local.address`

https://www.ibm.com/support/knowledgecenter/en/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/twlp_admin_localconnector.html

It sounds like if we were using that, we shouldn't actually need to use the attach API at all, though it's probably worth checking. (And also checking whether the current code works or not without localConnector)

**Comment by Azquelt** [comment]
Testing the existing code without your changes, localConnector-1.0 is required. Given that I'd suggest:

1. We remove the existing code which uses the attach API altogether and just use this file to find the address.
** I think we also use the attach API to see if the server is running, not sure if there's a better way to do that.

1. Even though liberty writes both files, we should use the one that's documented.

File: liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java [hunk 1]
```diff
@@ -322,13 +323,29 @@ private String getVMLocalConnectorAddress(VirtualMachine wlpvm)
...
+      
       serviceURL = wlpvm.getAgentProperties().getProperty(PROPERTY_NAME);
 
       // On some environments like the IBM JVM the localConnectorAddress is not
       // in the AgentProperties but in the SystemProperties.
       if (serviceURL == null)
          serviceURL = wlpvm.getSystemProperties().getProperty(PROPERTY_NAME);
+      
+      if (serviceURL == null) {
+          File jmxAddr = new File(getServerOutputDir() + "/workarea/com.ibm.ws.jmx.local.address");
```
Comment (NottyCode) [inline liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java]: you shouldn't be reading this file. Instead you should be looking at the equivalent file in the logs/state folder.

**Comment by aguibert** [comment]
I've updated my PR to look at `logs/state/` instead of `workarea/`. However, I hesitate to remove the existing Attach API code, because I want to have minimal impact on existing users.

**Comment by Azquelt** [comment]
Yeah, ok.

Looking at the docs though, I still think that not using the attach API and connecting to the URL from the file in `${server.output.dir}/logs/state/com.ibm.ws.jmx.local.address` is the more "correct" way to go about what we're doing here so I'll raise an issue to investigate doing that.

**Comment by Azquelt** [comment]
Can you rebase this? Then we can merge it.

**Pull request merged**